### ISSUE TITLE
Rate limit errors

### ISSUE THREAD

**Issue by danielpops** [Legit-Labs/legitify#110]
### TL;DR

When `legitify analyze`'ing across large number of repos (in  my organization, there are ~290), rate limit errors are encountered, and there is no way to slow the rate and mitigate the errors (other than reducing the number of repos in scope for `analyze`)

### Expected behavior

Exponential backoff by default and/or allow the caller to specify a value for wait between requests

### Observed behavior

Errors like this:
```
2023/01/12 11:09:25 2023/01/12 11:09:25 error getting branch protection info for the_repo: GET https://api.github.com/repos/TheOrg/the_repo/branches/master/protection: 403 You have exceeded a secondary rate limit. Please wait a few minutes before you try again.
```

which results in incomplete results in output.

### Version

legitify version 0.2.1 commit a5a45e1b4c4a737452d510e12c99feada658cb7b

### On which operating system are you using legitify?

Linux

### Relevant log output

```text
2023/01/12 11:09:25 2023/01/12 11:09:25 error getting branch protection info for the_repo: GET https://api.github.com/repos/TheOrg/the_repo/branches/master/protection: 403 You have exceeded a secondary rate limit. Please wait a few minutes before you try again.
```
```


### Additional information

_No response_

**Comment by gofri** [comment]
Sounds legit.
While we work it out, I'd suggest using list-repos to get the list of repos, then use the output to run analyze with the --repository flag and a subset of the repos each time (let me know if you need help with that)

**Comment by gal-legit** [comment]
@danielpops can you please checkout `gofri/transport`, run `make build` and run legitify to see if that fixes the issue for you?

**Comment by gal-legit** [comment]
hey @danielpops,
The secondary rate limit fix is now released: https://github.com/Legit-Labs/legitify/releases/tag/v0.2.2
so feel free to pick legitify off the release and try it out.
I'm closing this issue for now; If you still tackle rate limit issues running v0.2.2 please comment here and we'll reopen it.
